source/common/upstream/eds.cc

```diff
@@ -93,19 +93,25 @@ void EdsClusterImpl::onConfigUpdate(const ResourceVector& resources) {
     }
   }
 
+  // Track whether we rebuilt any LB structures.
+  bool cluster_rebuilt = false;
   for (size_t i = 0; i < priority_state.size(); ++i) {
     if (priority_state[i].first != nullptr) {
-      updateHostsPerLocality(priority_set_.getOrCreateHostSet(i), *priority_state[i].first,
-                             priority_state[i].second);
+      cluster_rebuilt |= updateHostsPerLocality(i, priority_set_.getOrCreateHostSet(i),
```

Comment:
can we either pass priority through and use it consistently in updateHostsPerLocality, or always pass the appropraite per-prioriy state rather than mixing and matching?